site/cypress/e2e/sidebar.spec.js

```diff
@@ -0,0 +1,83 @@
+import { getCurrentSemester } from '../support/utils.js';
+
+const currentSemester = `${getCurrentSemester()}`;
+
+Cypress.Commands.add('sidebarContains', (title, extension) => {
+    cy.get('aside ul li').contains(title).should('have.attr', 'href').and('contain', extension);
+    if (!title.includes('Logout')) {
+        cy.get('aside ul li').contains(title).click({ force: true });
+        if (title.includes('Autograding')) {
+            cy.get('#main > .content > h1').should('contain', 'Job Statistics');
+        }
+        else if (title.includes('Discussion Forum')) {
+            cy.get('#main > .content').should('contain', 'Create Thread');
+        }
+        else if (title.includes('Collapse Sidebar')) {
+            return;
+        }
+        else {
+            cy.get('#main > .content').should('contain', title);
+        }
+    }
+});
+
+Cypress.Commands.add('baseSidebar', (user) => {
+    cy.sidebarContains('My Courses', '/home');
+    cy.sidebarContains('My Profile', '/user_profile');
+    cy.sidebarContains('Authentication Tokens', '/authentication_tokens');
+    cy.sidebarContains('Calendar', '/calendar');
+    if (user.includes('instructor')) {
+        cy.sidebarContains('Docker', '/admin/docker');
+        cy.sidebarContains('New Course', '/home/courses/new');
+        cy.sidebarContains('Autograding Status', '/autograding_status');
+    }
```

Comment:
Why not have a separate command for the instructor sidebar items?  Doing that ensures that the tests get run every time, as opposed to the current solution which could become fragile if something cuts off this branch somehow.